nixos/modules/security/pam.nix

```diff
@@ -460,6 +460,94 @@ let
         description = lib.mdDoc "Contents of the PAM service file.";
       };
 
+      types = mkOption {
+        # This option is experimental and subject to breaking changes without notice.
+        visible = false;
+
+        description = lib.mdDoc ''
+          PAM rules for this service.
+
+          Attribute keys are the `type` of the rules (one of `account`, `auth`, `password`, `session`).
+        '';
+        type = types.attrsOf (types.submodule {
```

Comment:
My concern was that this would result in repetition in the manual. I guess there wouldn't be _that_ many repeated options, though. We can put it on my wishlist for the module system to support reusable types that are documented in one place in the manual. :smile:

Reply:
Indeed that would be nice, but for now I think it's fine with the duplication in the manual. This does have the benefit of making the documentation more obvious, and it also makes sure you can't have a typo in the key.

Reply:
Fixed in d6bb805932ae8fdea7e516f40b2455109d4b4013.